src/main/java/mods/railcraft/common/blocks/tracks/outfitted/kits/TrackKitMessenger.java

```diff
@@ -0,0 +1,74 @@
+/*------------------------------------------------------------------------------
+ Copyright (c) CovertJaguar, 2011-2016
+ http://railcraft.info
+
+ This code is the property of CovertJaguar
+ and may only be used with explicit written
+ permission unless otherwise specified on the
+ license page at http://railcraft.info/wiki/info:license.
+ -----------------------------------------------------------------------------*/
+package mods.railcraft.common.blocks.tracks.outfitted.kits;
+
+import mods.railcraft.common.blocks.tracks.outfitted.TrackKits;
+import net.minecraft.entity.item.EntityMinecart;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+
+import java.lang.ref.WeakReference;
+
+import javax.annotation.Nullable;
+
+public class TrackKitMessenger extends TrackKitRailcraft {
+
+    ITextComponent text = new TextComponentString("");
+    WeakReference<EntityMinecart> lastCart;
+    long lastTime;
+
+    @Override
+    public TrackKits getTrackKitContainer() {
+        return TrackKits.MESSENGER;
+    }
+
+    @Override
+    public boolean blockActivated(EntityPlayer player, EnumHand hand, @Nullable ItemStack heldItem) {
+        if (heldItem != null) {
+            if (heldItem.hasDisplayName()) {
+                text = new TextComponentString(heldItem.getDisplayName());
+            }
+        }
+        return super.blockActivated(player, hand, heldItem);
+    }
+
+    @Override
+    public void onMinecartPass(EntityMinecart cart) {
+        long time = cart.worldObj.getWorldTime();
+        if (lastCart != null && lastCart.get() == cart) {
```

Comment:
Probably will add a train mode that notifies all the riders of the train at the same time. How do you think?